paymentsheet/src/main/java/com/stripe/android/link/verification/LinkEmbeddedState.kt

```diff
@@ -0,0 +1,39 @@
+package com.stripe.android.link.verification
+
+import android.os.Parcelable
+import com.stripe.android.link.ui.verification.VerificationViewState
+import kotlinx.parcelize.Parcelize
+
+/**
+ * Unified state for Link embedded functionality including verification and payment method selection
+ */
+@Parcelize
+internal data class LinkEmbeddedState(
+    /**
+     * Current verification state representing the different stages in the verification process.
+     */
+    val verificationState: VerificationState,
+) : Parcelable
+
+/**
+ * Represents the different states of the verification process
+ */
+internal sealed class VerificationState : Parcelable {
```

Comment:
The naming here threw me off a bit, especially `Verifying`. Should we rename this to better describe what the outcome is, like `Loading`, `RenderInlineVerification`, and `RenderButton`?